compute_endpoint/globus_compute_endpoint/endpoint/utils/__init__.py

```diff
@@ -155,3 +156,28 @@ def send_endpoint_startup_failure_to_amqp(amqp_creds: dict, msg: str | None = No
                 body=payload,
                 mandatory=True,
             )
+
+
+def user_input_select(prompt: str, options: list[str]) -> str | None:
+    """
+    Prompt the user to select from a list of options, and returns the item or
+    None if the user declines to select one
+    """
+    assert options, "Options list must not be empty"
+    if not (sys.stdin and not sys.stdin.closed) or not sys.stdin.isatty():
+        raise ValueError("Not interactive, cannot prompt for user input")
+
+    print(prompt)
+    for idx, option_text in enumerate(options):
+        print(f"  [{idx}]: {option_text}")
+
+    while True:
+        input_raw = input("\nChoice: ")
+        if not input_raw.strip():
+            return None
+        else:
+            if input_raw.isdigit():
+                input_num = int(input_raw)
+                if 0 <= input_num < len(options):
+                    return options[input_num]
+            print(f"Invalid choice: {input_raw}")
```

Comment:
Since we're playing with awareness of the tty-connectedness, this might be a time to consider using colorama for the error handling.  This will play nice with auto colors if not to the tty.  For example:
```python
print(f"{colorama.Fore.RED}{colorama.Style.BRIGHT}Invalid choice:{colorama.Style.RESET_ALL} {input_raw}")

# or perhaps easier

hl = colorama.Fore.RED + colorama.Style.BRIGHT  # "highlight"
rs = colorama.Style.RESET_ALL
...
print(f"{hl}Invalid choice:{rs} {input_raw}")
```

Meanwhile, many CLI prompts will redraw the choices menu.  Consider, for example, `git add -p` and what happens with a typo.

Reply:
I'm gonna leave this one out as a potential future enhancement, since it's really open ended and can lead to a lot of back and forth (and therefore is deserving of its own PR)